src/fetchData.js

```diff
@@ -0,0 +1,21 @@
+import { getUnit } from './utils';
+
+const getWeather = async (long, lat) => {
+  const unit = getUnit().name;
+  const url = `http://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${long}&units=${unit}&APPID=4e1f41a544e8138e1382d0bff94cc988`;
+  const result = await fetch(url, { mode: 'cors' })
+    .then((response) => {
+      if (response.status === 200) return response.json();
+      return Promise.reject(new Error('Cannot fetch data'));
+    });
+
+  return result;
+};
+
+const currentLocation = () => new Promise(
```

Comment:
Please clarify to me how this function actually works. It is returning a `Promise` that has the usual callback (with resolve, reject arguments) but we are also passing those same `resolve and reject` into the return function call. Where did `navigator` come from? 

I am sure there's a good explanation for this. Though the mere fact that the particular function created this confusion is evidence that we might need to refactor this. It appears it is heavily dependent on `navigator` wherever this object is from. Also we need to ensure we are using `Promises` the right way. It may not be resulting in any exception but it might not be giving us the best result either. For instance if I type a non-existent city in the search box I don't get any error message from the UI telling me that I was searching for a non-existent data. Related to this I investigated the resulting "Response" for the request and it still has a status of 200.

Reply:
This is the HTML Geolocation API taken from [here](https://www.w3schools.com/html/html5_geolocation.asp), turned into a promise so we can fetch the weather data after we get the current location.